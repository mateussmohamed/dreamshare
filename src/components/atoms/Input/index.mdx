---
name: Input
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import Input from './index'

# Input
<PropsTable of={Input} />

<Playground>
  <Input />
</Playground>
