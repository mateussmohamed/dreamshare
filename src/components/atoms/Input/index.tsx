import styled from 'styled-components'

export const Input = styled.input`
  padding: 10px 15px;
  border-radius: 3px;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.3);
  background-color: #ffffff;
  border: 1px solid #d4d4d4;
  outline: 0;
  ${props => props.theme.typography.robotoStyle};
  font-size: 14px;
  @media only screen and (min-width: 320px) {
    margin: 15px 0;
  }

  @media only screen and (min-width: 768px) {
    margin: 0 15px;
    min-width: 300px;
  }
`
