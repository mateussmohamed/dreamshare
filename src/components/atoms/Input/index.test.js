import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Input from './index'

describe('[<Input />]', () => {
  const wrap = (props = {}) => shallow(<Input {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({})

    expect(wrapper).toMatchSnapshot()
  })
})
