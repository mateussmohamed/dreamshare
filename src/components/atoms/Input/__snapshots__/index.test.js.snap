// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Input />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Input-ooqq1g-0",
        "isStatic": false,
        "rules": Array [
          "padding:10px 15px;border-radius:3px;box-shadow:0 1px 1px 0 rgba(0,0,0,0.3);background-color:#ffffff;border:1px solid #d4d4d4;outline:0;",
          [Function],
          ";font-size:14px;@media only screen and (min-width:320px){margin:15px 0;}@media only screen and (min-width:768px){margin:0 15px;min-width:300px;}",
        ],
      },
      "displayName": "Input",
      "render": [Function],
      "styledComponentId": "Input-ooqq1g-0",
      "target": "input",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
